plugins/analysis/webinterface/plugin.go

```diff
@@ -1,19 +1,97 @@
 package webinterface
 
 import (
-	"github.com/iotaledger/goshimmer/plugins/analysis/webinterface/httpserver"
-	"github.com/iotaledger/goshimmer/plugins/analysis/webinterface/recordedevents"
+	"errors"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/gobuffalo/packr/v2"
+	"github.com/iotaledger/goshimmer/packages/shutdown"
+	"github.com/iotaledger/goshimmer/plugins/config"
+	"github.com/iotaledger/hive.go/daemon"
+	"github.com/iotaledger/hive.go/logger"
 	"github.com/iotaledger/hive.go/node"
+	"github.com/labstack/echo"
+	"golang.org/x/net/context"
+	"golang.org/x/net/websocket"
+)
+
+// PluginName is the name of the analysis server web interface plugin.
+const PluginName = "Analysis-WebInterface"
+
+var (
+	// Plugin is the plugin instance of the analysis server web interface plugin.
+	Plugin    = node.NewPlugin(PluginName, node.Disabled, configure, run)
+	log       *logger.Logger
+	engine    *echo.Echo
+	assetsBox = packr.New("Analysis_WebInterface_Assets", "./static")
 )
 
-// Configure configures the plugin.
-func Configure(plugin *node.Plugin) {
-	httpserver.Configure()
-	recordedevents.Configure(plugin)
+func configure(plugin *node.Plugin) {
+	log = logger.NewLogger(PluginName)
+
+	engine = echo.New()
+	engine.HideBanner = true
+	engine.HidePort = true
+
+	// we only need this special flag, because we always keep a packed box in the same directory
+	if config.Node.GetBool(CfgAnalysisWebInterfaceDev) {
+		engine.Static("/static", "./plugins/analysis/webinterface/static")
+		engine.File("/", "./plugins/analysis/webinterface/static/index.html")
+	} else {
+		for _, res := range assetsBox.List() {
+			engine.GET(filepath.ToSlash(filepath.Join("/static", res)), echo.WrapHandler(http.StripPrefix("/static", http.FileServer(assetsBox))))
+		}
+		engine.GET("/", index)
+	}
+
+	engine.GET("/datastream", echo.WrapHandler(websocket.Handler(dataStream)))
+	configureEventsRecording(plugin)
+}
+
+func run(_ *node.Plugin) {
+	log.Infof("Starting %s ...", PluginName)
```

Comment:
In the other plugin.run we are not saying `Starting ...`, but I guess it can't hurt :shrug:

Reply:
Removed